sdk/eventgrid/azure-eventgrid/README.md

```diff
@@ -1,21 +1,241 @@
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.eventgrid import EventGridPublisherClient
+
+topic_hostname = "https://<name>.<region>.eventgrid.azure.net"
+credential = AzureKeyCredential("<api_key>")
+eg_publisher_client = EventGridPublisherClient(topic_hostname, credential)
+```
+
+## Key concepts
+
+### EventGridPublisherClient
+`EventGridPublisherClient` provides operations to send event data to topic hostname specified during client initialization.
+Either a list or a single instance of CloudEvent/EventGridEvent/CustomEvent can be sent.
+
+### EventGridConsumer
+`EventGridConsumer` is used to desrialize an event received.
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common EventGrid tasks, including:
+
+* [Send an EventGrid Event](#send-an-eventgrid-event)
+* [Send a Cloud Event](#send-a-cloud-event)
+* [Consume an eventgrid Event](#consume-an-eventgrid-event)
+* [Consume a cloud Event](#consume-a-cloud-event)
+
+### Send an EventGrid Event
+
+This example publishes an EventGrid event.
+
+```Python
+import os
+from azure.core.credentials import AzureKeyCredential
+from azure.eventgrid import EventGridPublisherClient, EventGridEvent
+
+key = os.environ["EG_ACCESS_KEY"]
+topic_hostname = os.environ["EG_TOPIC_HOSTNAME"]
+
+event = EventGridEvent(
+    subject="Door1",
+    data={"team": "azure-sdk"},
+    event_type="Azure.Sdk.Demo",
+    data_version="2.0"
+)
+
+credential = AzureKeyCredential(key)
+client = EventGridPublisherClient(topic_hostname, credential)
+
+client.send(event)
+```
+
+### Send a Cloud Event
+
+This example publishes a Cloud event.
+
+```Python
+import os
+from azure.core.credentials import AzureKeyCredential
+from azure.eventgrid import EventGridPublisherClient, CloudEvent
+
+key = os.environ["CLOUD_ACCESS_KEY"]
+topic_hostname = os.environ["CLOUD_TOPIC_HOSTNAME"]
+
+event = CloudEvent(
+    type="Azure.Sdk.Sample",
+    source="https://egsample.dev/sampleevent",
+    data={"team": "azure-sdk"}
+)
+
+credential = AzureKeyCredential(key)
+client = EventGridPublisherClient(topic_hostname, credential)
+
+client.send(event)
+```
+
+### Consume an eventgrid Event
+
+This example demonstrates consuming and deserializing an eventgrid event.
+
+```Python
+import os
+from azure.eventgrid import EventGridConsumer
+
+consumer = EventGridConsumer()
+
+eg_storage_dict = {
+    "id":"bbab6625-dc56-4b22-abeb-afcc72e5290c",
+    "subject":"/blobServices/default/containers/oc2d2817345i200097container/blobs/oc2d2817345i20002296blob",
+    "data":{
+        "api":"PutBlockList",
+    },
+    "eventType":"Microsoft.Storage.BlobCreated",
+    "dataVersion":"2.0",
+    "metadataVersion":"1",
+    "eventTime":"2020-08-07T02:28:23.867525Z",
+    "topic":"/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/t-swpill-test/providers/Microsoft.EventGrid/topics/eventgridegsub"
```

Comment:
(I realize in hindsight we may want to anonymize some of this away from Swathi's bits so as to not confuse folks)